DeviceSDK/src/main/java/org/astarteplatform/devicesdk/util/UUID5.java

```diff
@@ -0,0 +1,61 @@
+package org.astarteplatform.devicesdk.util;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * This class is based on stackoverflow question, with some changes.
+ *
+ * @see <a
```

Comment:
I would keep this as part of the file header rather than in our javadoc, it is not really useful for API users, while it is an important information from a contributors perspective.